broker/submissions/routes.py

```diff
@@ -16,23 +17,45 @@
 )
 
 
+@submissions_bp.route('/<submission_uuid>/spreadsheet', methods=['POST'])
+def generate_spreadsheet(submission_uuid):
+    submission = app.ingest_api.get_submission_by_uuid(submission_uuid)
+    spreadsheet_job = submission.get('lastSpreadsheetGenerationJob', {}) or {}
+
+    message = 'The spreadsheet is being generated.'
+    if not spreadsheet_job.get('createdDate') or (spreadsheet_job.get('createdDate') and spreadsheet_job.get('finishedDate')):
+        spreadsheet_export_service = ExportToSpreadsheetService(app.ingest_api)
+        spreadsheet_export_service.async_export_and_save(submission_uuid, app.SPREADSHEET_STORAGE_DIR)
+        return response_json(HTTPStatus.ACCEPTED, {'message': message})
+    else:
+        return response_json(HTTPStatus.ACCEPTED, {'message': message})
+
+
 @submissions_bp.route('/<submission_uuid>/spreadsheet', methods=['GET'])
-def export_to_spreadsheet(submission_uuid):
-    workbook = ExportToSpreadsheetService(app.ingest_api).export(submission_uuid)
-    timestamp = time.strftime("%Y%m%d-%H%M%S")
-    temp_file = tempfile.NamedTemporaryFile()
-    filename = f'{submission_uuid}_{timestamp}.xlsx'
-    workbook.save(temp_file.name)
-    return send_file(temp_file.name,
-                     as_attachment=True,
-                     cache_timeout=0,
-                     attachment_filename=filename)
+def download_spreadsheet(submission_uuid):
+    submission = app.ingest_api.get_submission_by_uuid(submission_uuid)
+    spreadsheet_job = submission.get('lastSpreadsheetGenerationJob', {}) or {}
+    if spreadsheet_job.get('finishedDate', {}):
+        create_date = parse_date_string(spreadsheet_job.get('createdDate'))
+        timestamp = create_date.strftime("%Y%m%d-%H%M%S")
+        directory = f'{app.SPREADSHEET_STORAGE_DIR}/{submission_uuid}'
+        filename = f'{submission_uuid}_{timestamp}.xlsx'
+        filepath = f'{directory}/downloads/{filename}'
```

Comment:
This seems like the same logic as in `export_to_spreadsheet_service.py`

Reply:
I've extracted the logic of constructing the directory, filename and filepath and place them all as static method in the service.